### PR TITLE
Add Connection test for Api Pull Notification request

### PR DESCRIPTION
**Pull request by puntope**
### Changes proposed in this Pull Request:

This PR adds a section in the connection test page for Notifying an item with a topic.

### Screenshots:

Screenshot 2024-01-26 at 09 47 26

### Detailed test instructions:

1. Go to Connection Test page
2. See the new section Partner API Pull Integration
3. Send a request with a number and a topic

### Additional details:

- Validation of item (product/copuon) will be done in future PRs

### CODE REVIEW

File: src/ConnectionTest.php [hunk 1]
```diff
@@ -9,12 +9,14 @@
 
 namespace Automattic\WooCommerce\GoogleListingsAndAds;
 
+use Automattic\Jetpack\Connection\Client;
```
Comment (jorgemd24) [inline src/ConnectionTest.php]: Unused `Automattic\Jetpack\Connection\Client`

File: src/ConnectionTest.php [hunk 2]
```diff
@@ -30,6 +32,7 @@
 use Automattic\WooCommerce\GoogleListingsAndAds\Product\ProductSyncer;
 use Automattic\WooCommerce\GoogleListingsAndAds\Product\ProductSyncerException;
 use Automattic\WooCommerce\GoogleListingsAndAds\Vendor\Psr\Container\ContainerInterface;
+use Jetpack_Client;
```
Comment (jorgemd24) [inline src/ConnectionTest.php]: Unused Jetpack_Client

File: src/ConnectionTest.php [hunk 3]
```diff
@@ -625,6 +628,43 @@ protected function render_admin_page() {
...
+									<label>
+										Topic
+										<select name="topic">
+											<option value="product.create" <?php echo (! isset( $_GET['topic'] ) || $_GET['topic'] === 'product.create') ? "selected" : "" ?>>product.create</option>
+											<option value="product.delete" <?php echo $_GET['topic'] === 'product.delete' ? "selected" : ""?>>product.delete</option>
+											<option value="product.update" <?php echo $_GET['topic'] === 'product.update' ? "selected" : ""?>>product.update</option>
+											<option value="coupon.create" <?php echo $_GET['topic'] === 'coupon.create' ? "selected" : ""?>>coupon.create</option>
+											<option value="coupon.delete" <?php echo $_GET['topic'] === 'coupon.delete' ? "selected" : ""?>>coupon.delete</option>
+											<option value="coupon.update" <?php echo $_GET['topic'] === 'coupon.update' ? "selected" : ""?>>coupon.update</option>
+										</select>
```
Comment (jorgemd24) [inline src/ConnectionTest.php:653]: Can we add the topics for shippings?

https://github.com/woocommerce/google-listings-and-ads/blob/8b273bc98de2bf05c6a48f1f5e688622d1e3da85/src/Google/NotificationsService.php#L27-L28

**Review by jorgemd24 — COMMENTED** [review]
@puntope Thanks for adding this,  I just noticed that we're not handling the errors returned from WPCOM when the notification fails. Currently, it always returns success regardless of the status code.

https://github.com/woocommerce/google-listings-and-ads/blob/8b273bc98de2bf05c6a48f1f5e688622d1e3da85/src/Google/NotificationsService.php#L68-L72

Let me know if you want to address this issue in this PR or if you'd prefer to tackle it in a different PR.

File: src/ConnectionTest.php (hunk 3 above)
Comment (puntope) [reply src/ConnectionTest.php:653]: This I will also add it when working on Shipping and Settings if it's ok

**Comment by puntope** [comment]
> Let me know if you want to address this issue in this PR or if you'd prefer to tackle it in a different PR.

I will check this and work on it in another PR if that's ok @jorgemd24

**Comment by puntope** [comment]
Thanks @jorgemd24 for the review. I addressed the suggestion concerning the dependencies. 

The other 2 issues I will handle it separately if that's ok. This is ready for a new round. Thanks

**Review by jorgemd24 — APPROVED** [review]
Thanks @puntope , LGTM 👍

**Comment by puntope** [comment]
> I just noticed that we're not handling the errors returned from WPCOM when the notification fails. Currently, it always returns success regardless of the status code.

Hi @jorgemd24 I tried to reproduce the issue but I couldn't.

We do check for WP errors as well as status codes.

Screenshot 2024-02-06 at 12 12 12

Screenshot 2024-02-06 at 12 11 54

Can you help me to reproduce the issue?

**Pull request merged**